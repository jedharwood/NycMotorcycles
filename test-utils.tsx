import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import en from './languages/en.json';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import { QueryClient, QueryClientProvider } from 'react-query';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const queryClient = new QueryClient();

  return (
    <MemoryRouterProvider>
      <QueryClientProvider client={queryClient}>
        <IntlProvider locale='en' messages={en}>
          {children}
        </IntlProvider>
      </QueryClientProvider>
    </MemoryRouterProvider>
  );
};

const renderWithProviders = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { renderWithProviders as render };
