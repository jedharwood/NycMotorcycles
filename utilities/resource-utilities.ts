export const yahooAuctionProfilePage: string =
  'https://auctions.yahoo.co.jp/seller/lazzamoto?'

export const instagramProfilePage: string =
  'https://www.instagram.com/newyorkcitymotorcycles/'

export const routes: { [key: string]: string } = {
  home: '/',
  activeAuctions: '/active-auctions',
  soldArchive: '/sold-archive',
  history: '/history',
  racing: '/racing',
  contact: '/contact',
  consign: '/consign',
  invest: '/invest',
  download: '/download',
}

const gallery: string = '/sold-archive/gallery/'
export const galleryRoutes: { [key: string]: string } = {
  hondaRc30: `${gallery}honda-rc30`,
  suzukiRg400: `${gallery}suzuki-rg400`,
  shovelheadChopper: `${gallery}shovelhead-chopper`,
  laverda750Sfc: `${gallery}laverda-750-sfc`,
  harleyXr750: `${gallery}harley-xr750`,
  kawasakiH2: `${gallery}kawasaki-h2`,
  suzukiStinger: `${gallery}suzuki-stinger`,
  bsaStarfire: `${gallery}bsa-starfire`,
  kawasakiH1: `${gallery}kawasaki-h1`,
}
