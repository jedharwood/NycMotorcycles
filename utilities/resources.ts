export const instagramLink: string =
    'https://www.instagram.com/newyorkcitymotorcycles/';
