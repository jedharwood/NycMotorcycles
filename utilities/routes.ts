const routes: { [key: string]: string } = {
  home: '/',
  activeAuctions: '/active-auctions',
  soldArchive: '/sold-archive',
  history: '/history',
  racing: '/racing',
  contact: '/contact',
  consign: '/consign',
  invest: '/invest',
  download: '/download',
  gallery: '/sold-archive/gallery',
};

export default routes;
