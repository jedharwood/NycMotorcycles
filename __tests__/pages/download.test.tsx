import DownloadPage from '@/pages/download';
import { render } from '@/test-utils';
import { langs } from '@/utilities/resources';

describe('DownloadPage', () => {
    it('should render page in English', () => {
        const { container } = render(<DownloadPage />);

        expect(container).toMatchSnapshot();
    });

    it('should render page in Japanese', () => {
        const { container } = render(<DownloadPage />, { locale: langs.ja });

        expect(container).toMatchSnapshot();
    });
});
