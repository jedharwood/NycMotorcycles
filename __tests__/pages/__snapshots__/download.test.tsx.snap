// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadPage should render page in English 1`] = `
<div>
  <main
    class="space-y-6"
  >
    <section
      class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
    >
      <div
        class="text-center "
      >
        <h2
          class="text-xl font-medium opacity-80 md:text-2xl"
        >
          Power of Attorney form
        </h2>
      </div>
      <div
        class="space-y-4 undefined"
      >
        <p>
          The power of attorney form permits our shipping agent to move your bike internationally.
        </p>
        <p>
          Please contact us if you need assistance in completing this form; we're happy to help.
        </p>
      </div>
      <div
        class="flex justify-center"
      >
        <a
          download="mock-form-one"
          href="/documents/mock-form-one.pdf"
        >
          <div
            class="flex justify-center"
          >
            <button
              class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
              type="button"
            >
              Download
            </button>
          </div>
        </a>
      </div>
    </section>
    <section
      class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
    >
      <div
        class="text-center "
      >
        <h2
          class="text-xl font-medium opacity-80 md:text-2xl"
        >
          Export Invoice form
        </h2>
      </div>
      <div
        class="space-y-4 undefined"
      >
        <p>
          The export invoice form will detail the value of your motorcycle and is required in order for your bike to clear customs here in Japan.
        </p>
        <p>
          Please contact us if you need assistance in completing this form; we're happy to help.
        </p>
      </div>
      <div
        class="flex justify-center"
      >
        <a
          download="mock-form-two"
          href="/documents/mock-form-two.pdf"
        >
          <div
            class="flex justify-center"
          >
            <button
              class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
              type="button"
            >
              Download
            </button>
          </div>
        </a>
      </div>
    </section>
  </main>
</div>
`;
